### DISCUSSION TITLE
Verifying Compatibility with TestConsole

### DISCUSSION THREAD

**Discussion by Mick3DIY** [fasteddy516/CircuitPython_JoystickXL#21] [Q&A] [answered]
Hello there ! 

I'm new in the Raspi Pico world and I would like to use the circuitPython JoystickXL to build a tiny box for flying in some simulators...

I follow the first part : download the last version, copy/paste in the Pico lib folder, create a boot.py file...

When I try to run the testConsole command, I have a error during the import :

![JoystickXL_01](https://github.com/fasteddy516/CircuitPython_JoystickXL/assets/42770997/49db7e6b-e19f-4a98-8f1f-ae626b1591d9)

I'm using Ubuntu 22 with Thonny IDE. When I copy and paste tools.py in the lib/joystick_xl, it works well...

![JoystickXL_02](https://github.com/fasteddy516/CircuitPython_JoystickXL/assets/42770997/71b24054-6a67-4695-a3a8-c79bea8f63da)

What I missed or where I have make a mistake ? Related to this commit [Prevent TestConsole failure on board with no D2 pin defined](https://github.com/fasteddy516/CircuitPython_JoystickXL/commit/f597db5f3fae2cec89cce8b5c7cf9a64df90694b) ?

Thank you & have nice week ! Cheers from France.

Mike

**Comment by fasteddy516** [comment, marked as answer]
You're absolutely correct - the commit that fixes `TestConsole` on boards without a `D2` pin is the difference here.  The latest released `.mpy` files are lagged behind the current source `.py` files by a couple of commits at the moment.  I'll be fixing this soon, but for now using the `tools.py` file as you have should work fine.

Note that - using `tools.py` - you can specify a different pin to use for TestConsole from the REPL:
```
>>> import board
>>> from joystick_xl.tools import TestConsole
>>> TestConsole(button_pin=board.GP0)
```

Hopefully that helps.  I'll post back here when I've added a new release that brings the `.mpy` files up-to-date.

**Reply by Mick3DIY** [reply]
Thank you for your help ! Next step, read the doc, read the (entire) doc ! :)

**Comment by fasteddy516** [comment]
I've just published a minor update (0.4.2) that brings the release files up-to-date.

**Comment by Mick3DIY** [comment]
Thanks again @fasteddy516 for the update, I have updated a educational project with your library : [https://github.com/Mick3DIY/EasyPicoPCB/tree/main/joystickXL](https://github.com/Mick3DIY/EasyPicoPCB/tree/main/joystickXL)

It works well ! Next step I will make a simple multibox (buttons/switches, potentiometers, leds...)  for DCS, ETS2... :beers:
